addons/hr_holidays/models/hr_leave_type.py

```diff
@@ -116,8 +117,8 @@ def _search_valid(self, operator, value):
             or that don't need an allocation
             return [('id', domain_operator, [x['id'] for x in res])]
         """
-        date_to = self._context.get('default_date_from') or fields.Date.today().strftime('%Y-1-1')
-        date_from = self._context.get('default_date_to') or fields.Date.today().strftime('%Y-12-31')
+        date_to = pytz.utc.localize(datetime.datetime.fromisoformat(self._context.get('default_date_from'))).astimezone(pytz.timezone(self._context.get('tz'))).strftime('%Y-%m-%d') if self._context.get('default_date_from') else fields.Date.today().strftime('%Y-1-1')
+        date_from = pytz.utc.localize(datetime.datetime.fromisoformat(self._context.get('default_date_to'))).astimezone(pytz.timezone(self._context.get('tz'))).strftime('%Y-%m-%d') if self._context.get('default_date_to') else fields.Date.today().strftime('%Y-12-31')
```

Comment:
Maybe break this down in multiple lines to make it a bit more readable and make shorter lines :+1:

Reply:
yup, major rework done

PS you were very quick to stalk out my draft version :laughing: